src/components/EventModal.vue

```diff
@@ -0,0 +1,108 @@
+<template>
+  <div class="popup">
+    <div class="popup-content" id="popup-content">
+      <button class="close" @click="close">&times;</button>
+      <h2>{{ title }}</h2>
+      <br />
+      <p>{{ desc }}</p>
+      <div v-if="past">
+        <p>This event occurred on {{ dateString }}.</p>
+      </div>
+      <div v-else>
+        <div class="form-item">
+          <label>Enter your check-in code:</label>
+          <br />
+          <input
+            type="text"
+            v-model.number="code"
+            maxlength="4"
+            input-code
+            class="code"
+            placeholder="1234"
+          />
+        </div>
+        <br />
+        <button
+          class="form-item"
+          :disabled="!codeIsValid"
+          @click="submitCheckIn()"
+        >
+          Check in
+        </button>
+        <p v-if="!codeIsValid" class="error" error-code>
+          Please enter valid 4 digit code
+        </p>
+        <p v-if="badCode" class="error" error-code>
+          The code you entered was incorrect.
+        </p>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "ListCard",
+
+  props: {
+    event: {
+      validator: prop => typeof prop === "object" || prop === null,
```

Comment:
Is there a way to "shape" the prop type? eg require certain fields of a prop too. I've done this in react but not sure about vue